packages/react-reconciler/src/ReactInternalTypes.js

```diff
@@ -259,6 +259,8 @@ type BaseFiberRootProperties = {
   pooledCache: Cache | null,
   pooledCacheLanes: Lanes,
 
+  forceSync: boolean,
```

Comment:
It's not really a "forced" sync because it just an best-effort attempt. This could easily be confused with the API called `ReactDOM.flushSync(...)` which is more like a force.

This would be more like `attemptEagerTransition`. I don't think you should necessarily need this flag though with the new refactoring because once we schedule an eager microtask all the time, you can just do the work right there if you check for it.

```
if (shouldAttemptEagerTransition()) performWorkOnRoot()
```

A lot of things will be a lot simpler with that model.